src/script/engine.zig

```diff
@@ -312,6 +314,14 @@ pub const Engine = struct {
         try self.stack.pushByteArray(first);
     }
 
+    /// OP_2ROT: The fifth and sixth items back are moved to the top of the stack
+    fn op2Rot(self: *Engine) !void {
+        try self.stack.swap(0, 2);
+        try self.stack.swap(1, 3);
+        try self.stack.swap(2, 4);
+        try self.stack.swap(3, 5);
```

Comment:
index 0 is the item at the very bottom of the stack. 
The item at the top is stack.len -1;

You test only pass because your input stack is 6 element log. Add some at the bottom and it will fail

Reply:
@tdelabro can you check plz from this [way i used in the first](https://github.com/zig-bitcoin/btczee/pull/143/commits/8ea519d3fec4375ea55dc99f48a7bc1186a0f5b6)